Lib/typing.py

```diff
@@ -440,8 +442,32 @@ def NoReturn(self, parameters):
       def stop() -> NoReturn:
           raise Exception('no way')
 
-    This type is invalid in other positions, e.g., ``List[NoReturn]``
-    will fail in static type checkers.
+    Static type checkers treat this as a general bottom type,
+    a type with no members. The typing.Never type provides a
+    more explicit name for that concept.
+
+    """
+    raise TypeError(f"{self} is not subscriptable")
+
+@_SpecialForm
+def Never(self, parameters):
+    """Notation for the bottom type, a type that has no members.
+
+    This can be used to define a function that should never be
+    called, or a function that never returns::
+
+      from typing import Never
+
+      def never_call_me(arg: Never) -> None:
+          pass
+
+      never_call_me(1)  # type checker error
+
+      def stop() -> Never:
+          return 1  # type checker error
+
+    The assert_never() function uses the Never type to statically
+    assert that code is unreachable.
```

Comment:
Does the docstring need to say this? Does assert_never() need this much advertising? :-)